components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2445,4 +2445,96 @@ void RewardsServiceImpl::GetAddressesForPaymentId(
                      callback));
 }
 
+brave_rewards::PendingContributionInfoList PendingContributionsOnFileTaskRunner(
+    PublisherInfoDatabase* backend) {
+  brave_rewards::PendingContributionInfoList list;
+  if (!backend) {
+    return list;
+  }
+
+  backend->GetPendingContributions(&list);
+
+  return list;
+}
+
+void RewardsServiceImpl::GetPendingContributions(
+    const GetPendingContributionsCallback& callback) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&PendingContributionsOnFileTaskRunner,
+                 publisher_info_backend_.get()),
+      callback);
+}
+
+bool RemovePendingContributionOnFileTaskRunner(
+    PublisherInfoDatabase* backend,
+    const std::string& publisher_key,
+    const std::string& viewing_id,
+    uint64_t added_date) {
+  if (!backend) {
+    return false;
+  }
+
+  return backend->RemovePendingContributions(publisher_key,
+                                             viewing_id,
+                                             added_date);
+}
+
+void RewardsServiceImpl::RemovePendingContribution(
+    const std::string& publisher_key,
+    const std::string& viewing_id,
+    uint64_t added_date) {
+  base::PostTaskAndReplyWithResult(
```

Comment:
so like we talked about in DM, these should still go through ledger lib. The db persistence will change in the future to put control in the hands of the ledger lib and in general we don't want to bypass it like this

Reply:
done